content/topics/Analyze/machine-learning/ml-intro/supervised-learning-101.md

```diff
@@ -0,0 +1,333 @@
+{{% /tip %}}
+
+## A Classification Example: Decision Trees
+In this example of supervised classification in R we will use the `iris` dataset. This dataset consists of measurements for iris flowers from three different species. We will train a model that can classify the irises into one of the three speciess. We will do this by making a decision tree.
+
+### Loading and Inspecting the Data
+First, load the iris dataset and inspect what it looks like:
+
+{{% codeblock %}}
+```R
+# Load the iris dataset
+library(datasets)
+data("iris")
+df <- iris # call the dataframe `df`
+
+# Inspect the data
+summary(df)
+plot(df)
+
+{{% /codeblock %}}
+
+This data contains five columns with flower measurements. There are observations of 150 flowers. 
+
+### Splitting the data
+For classification, we also split the data into a training set and a testing set. We again specify how large we want the split to be, and randomly assign rows to one of the two dataframes `df_train` or `df_test`:
+
+{{% codeblock %}}
+```R
+# Splitting the data
+set.seed(123) 
+N <- nrow(df) # Number of rows in df
+training_rows <- sample(1:N, 0.80*N) # 80-20 random split
+
+df_train <- df[training_rows, ] # Training data
+df_test <- df[-training_rows, ] # Testing data
+
+# Verify the number of rows
+nrow(df_train)
+nrow(df_test)  
+{{% /codeblock %}}
+
+### Training the Model
+We will use a decision tree to classify flowers in one of three species.  `Species` is the target and all other columns `.` are the features. 
+We only use the training data in `df_train`:
+
+{{% codeblock %}}
+```R
+# install.packages("rpart") # Uncomment if 'rpart' is not installed
+library(rpart)
+
+# Train the model
+model <- rpart(Species ~ ., data = df_train, method = "class")
+
+# Visualize the decision tree
+# install.packages("rpart.plot") # Uncomment if 'rpart.plot' is not installed
+library(rpart.plot)
+rpart.plot(model)
+
+{{% /codeblock %}}
+
+<p align = "center">
+<img src = "../images/classification_example_plot.png" width="500">
+</p>
+
+{{% tip %}}
+If you are using decision trees for a regression problem, you should adapt the method to "anova": 
+
+`model <- rpart(Y ~ ., data = df_train, method = "anova")`
+{{% /tip %}}
+
+### Making Predictions
+After training the model, we will make predictions using the `df_test` data. These values are assigned to a new column in the testing dataframe called `Species_pred`. After, we inspect the results by printing the predictions against the actual values in `Species`:
+
+{{% codeblock %}}
+```R
+# Predict values and assign to new column `Species_pred`
+df_test$Species_pred <- predict(model, newdata = df_test, type = "class")
+
+# Print the results
+df_test[c("Species", "Species_pred")]
+{{% /codeblock %}}
+
+### Evaluating the Model
+Now that we have trained and tested the model, we want to know how well it performs. 
+We look at the confusion matrix, which shows how many observations of each category are correctly classified:
```

Comment:
I would add hyperlink to a website explaining further the confusion matrix, for enhanced reference.